drivers/tty/serial/8250/8250_of.c

```diff
@@ -335,6 +353,12 @@ static const struct of_device_id of_platform_serial_table[] = {
 	{ .compatible = "ti,da830-uart", .data = (void *)PORT_DA830, },
 	{ .compatible = "nuvoton,wpcm450-uart", .data = (void *)PORT_NPCM, },
 	{ .compatible = "nuvoton,npcm750-uart", .data = (void *)PORT_NPCM, },
+#ifdef CONFIG_SERIAL_8250_NI16550
+	{ .compatible = "ni16550-fifo16",
```

Comment:
These don't actually match up with OF compatible naming due to lacking the vendor name-- `ni,ni16550-fifo16` or `ni,ni16550-fifo128` would be more correct-- I suspect we didn't have the `ni` vendor ID in the list at the time we originally wrote this.

Though that's probably a thing that can be pushed off to "whenever we make another pass at trying to make this upstream-friendly".

Reply:
Good catch. Let's revisit this as part of the upstreaming clean-up work since this branch is x86_64 only and it doesn't use device trees (and hence this code path is unused/untested).